src/nanoemoji/svg.py

```diff
@@ -557,27 +558,34 @@ def _ensure_groups_grouped_in_glyph_order(
 
     # The glyph names in the TTFont may have been dropped (post table 3.0), so the
     # names we see after decompiling the TTFont are made up and likely different
-    # from the input color glyph names. We only want to reorder the glyphs while
-    # keeping the existing names, we can't change order and rename at the same time
+    # from the input color glyph names. We only want to reorder the base color glyphs
+    # while keeping the current names: we can't change order and rename at the same time
     # or else tables that contain mappings keyed by glyph name would blow up.
     # Thus, we need to match the old and current names by their position in the
-    # font's current glyph order: i.e. we assume all color glyphs are placed at the
-    # END of the glyph order.
+    # font's current glyph order: i.e. we assume all color glyphs are placed in a
+    # continuous block starting at the first color glyph.
     current_glyph_order = ttfont.getGlyphOrder()
-    current_color_glyph_names = current_glyph_order[-len(color_glyphs) :]
+    first_color_gid = min(g.glyph_id for g in color_glyphs.values())
```

Comment:
Optional, you could do this in a reduce and get back a tuple (min, max). The slice below could then go from [min:max+1]

Reply:
simply taking min and max is much faster than doing a reduce in single pass